server/drivers/clickhouse/index.js

```diff
@@ -70,7 +70,21 @@ async function runQuery(query, connection) {
     },
   });
 
-  rows = await clickhouse.query(limitedQuery + ' FORMAT JSON').toPromise();
+  // clickhouse package will append ' FORMAT JSON' to certain queries,
+  // but does not handle CTEs WITH... SELECT.
+  // This borrows from approach used in package:
+  // https://github.com/TimonKK/clickhouse/blob/9dea3c0c3e4f3e2fe64e59dad762f1db044bf9bf/index.js#L479
+  // This is a bit naive but should cover most cases
+  if (
+    limitedQuery.trim().match(/^(with|select|show|exists)/i) &&
```

Comment:
ref: Issue #990 , this line means the sql need start with `with|select|show|exists`, but when I comment the first line, the error occurs.

Reply:
It is resolved.